OIIInspector/image_manager.py

```diff
@@ -0,0 +1,198 @@
+
+    def _pull_image(self):
+        """
+        Method that will pull the image with available container platform
+        """
+        run_cmd(f"{self._container_platform} pull {self._image_address}")
+
+    def _serve_index_registry(self):
+        """
+        Locally start podman/docker image service, which can be communicated with using gRPC queries.
+        Resolution of port conflicts is handled in this function.
+
+        :return: tuple containing port number of the running service and the running Popen object.
+        :rtype: (int, Popen)
+        :raises OIIInspectorError: if all tried ports are in use, or the command failed for another reason.
+        """
+        port_start = self._grpc_start_port
+        port_end = port_start + self._grpc_max_port_tries
+
+        for port in range(port_start, port_end):
+            try:
+                return (
+                    port,
+                    self._serve_index_registry_at_port(
+                        port, self._grpc_init_wait_time),
+                )
+            except AddressAlreadyInUse:
+                log.info('Port %d is in use, trying another.', port)
+
+        err_msg = "No free port has been found after {number_of_ports} attempts.".format(
+            number_of_ports=self._grpc_max_port_tries)
+        log.error(err_msg)
+        raise Exception(err_msg)
+
+    @retry(exceptions=OIIInspectorError, tries=2, logger=log)
+    def _serve_index_registry_at_port(self, port, wait_time):
+        """
+        Start an image registry service at a specified port.
+
+        :param str int port: port to start the service on.
+        :param wait_time: time to wait before checking if the service is initialized.
+        :return: object of the running Popen process.
+        :rtype: Popen
+        :raises OIIInspectorError: if the process has failed to initialize too many times, or an unexpected
+        error occured.
+        :raises AddressAlreadyInUse: if the specified port is already being used by another service.
+        """
+        cmd = f"{self._container_platform} run --name={self._base_container_name}_{port} " \
+              f"-p={port}:{port} {self._image_address}"
+
+        rpc_proc = subprocess.Popen(
+            shlex.split(cmd),
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            universal_newlines=True,
+        )
+        start_time = time.time()
+        while time.time() - start_time < wait_time:
+            time.sleep(1)
+            ret = rpc_proc.poll()
+            # process has terminated
+            if ret is not None:
+                stderr = rpc_proc.stderr.read()
+                print(stderr)
+                if "address already in use" in stderr:
+                    raise AddressAlreadyInUse(
+                        f"Port {port} is already used by a different service")
+                if "the container name" in stderr and "is already in use by" in stderr:
```

Comment:
I think the correct solution would be to simply increase the port number and try again. Otherwise, it wouldn't be possible to run multiple index images on the same machine at the same time.